### PR TITLE
Optimize the benchmark workflow

### PR DESCRIPTION
**Pull request by grief8**
This pull request includes changes to the `test/Makefile` and `test/benchmark/common/prepare_host.sh` files to improve the initialization and preparation processes for benchmarks. 

Improvements to initialization and preparation processes:

* [`test/Makefile`](diffhunk://#diff-740cb5a1689091cb894445e46683c255e39689ccba1c6c63d8a8841c8df8817dL28-R45): Modified the `INITRAMFS_ALL_DIRS` definition to conditionally include the test directory based on the `BENCHMARK` variable. The `apps` is no longer compiled and packed when we do the benchmarks.
* [`test/benchmark/common/prepare_host.sh`](diffhunk://#diff-bb76958e6a2d3a9cb54b58113057b18cd1f338d4a685e3c4258d80bdb0af7523R46-R49): Added a check to ensure the `initramfs.cpio.gz` file does not exist before running the `make initramfs` command.

### CODE REVIEW

File: test/benchmark/common/prepare_host.sh [hunk 1]
```diff
@@ -43,5 +43,8 @@ prepare_libs() {
 prepare_fs() {
     # Disable unsupported ext2 features of Asterinas on Linux to ensure fairness
     mke2fs -F -O ^ext_attr -O ^resize_inode -O ^dir_index ${BENCHMARK_ROOT}/../build/ext2.img
-    make initramfs
+    # Check if ${BENCHMARK_ROOT}/../build/initramfs.cpio.gz exists
+    if [ ! -f "${BENCHMARK_ROOT}/../build/initramfs.cpio.gz" ]; then
```
Comment (tatetian) [inline test/benchmark/common/prepare_host.sh:47]: Remove such worthless comments. Reading the comment `# Check if ${BENCHMARK_ROOT}/../build/initramfs.cpio.gz exists` does not save people any time than doing the code `if [ ! -f "${BENCHMARK_ROOT}/../build/initramfs.cpio.gz" ]; then`.

File: test/benchmark/common/prepare_host.sh [hunk 2]
```diff
@@ -43,5 +43,8 @@ prepare_libs() {
...
+        make initramfs
```
Comment (tatetian) [inline test/benchmark/common/prepare_host.sh:48]: Why the `initramfs` Makefile target may be skipped, but not `mke2fs` on Ext2? And what if any test changes? The initramfs image would NOT be re-generated. We want to have "optimization"s to speed up the benchmark, but not a half-based optimization that solve one problem but introduce another.

File: test/Makefile [hunk 3]
```diff
@@ -25,17 +25,24 @@ INITRAMFS_EMPTY_DIRS := \
...
 	$(INITRAMFS)/bin \
 	$(INITRAMFS)/sbin \
 	$(INITRAMFS)/usr/bin \
 	$(INITRAMFS)/usr/local \
-	$(INITRAMFS)/test \
 	$(INITRAMFS)/benchmark \
 	$(INITRAMFS_EMPTY_DIRS)
+
+# Conditionally include test directory if BENCHMARK is not 'none'
+ifneq ($(BENCHMARK), none)
```
[the remark this answers is not in the document]
Comment (grief8) [reply test/Makefile]: The benchmark running scripts are stored under the directory `$(initramfs)/test`.

The following rule ensures that the benchmark files are copied to the `$(INITRAMFS)/benchmark` directory, provided that the `$(INITRAMFS)/benchmark/bin` directory exists:

```makefile
$(INITRAMFS)/benchmark: | $(INITRAMFS)/benchmark/bin
	@cp -rf $(CUR_DIR)/benchmark/* $@
```

We do not want the `$(INITRAMFS)/test` target to be compiled when running benchmarks as follows:

```makefile
$(INITRAMFS)/test:
	@$(MAKE) --no-print-directory -C apps
```

We should avoid building the `test` target if the `BENCHMARK` variable is set to a value other than the default value `none`. This ensures that the benchmarks are executed without unnecessary builds. So it should be 
```makefile
ifeq ($(BENCHMARK), none)
INITRAMFS_ALL_DIRS += $(INITRAMFS)/test
endif
```

File: test/benchmark/common/prepare_host.sh (hunk 2 above)
Comment (grief8) [reply test/benchmark/common/prepare_host.sh]: You are right.

**Comment by grief8** [comment]
@boterinas ready

**Review by tatetian — APPROVED** [review]
LGTM. Thanks for the contribution!
